pkg/resourceinterpreter/configurableinterpreter/configmanager/accessor.go

```diff
@@ -0,0 +1,97 @@
+package configmanager
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+)
+
+// CustomConfiguration provides base information about custom interpreter configuration
+type CustomConfiguration interface {
+	GetConfigurationName() string
+	GetConfigurationTargetGVK() schema.GroupVersionKind
+}
+
+// LuaScriptAccessor provides a common interface to get custom interpreter lua script
+type LuaScriptAccessor interface {
```

Comment:
I'm thinking if we will supporting more scripts. So the `Lua` name is limited.

Reply:
Any suggestion?

Reply:
```
type ScriptAccessor interface {
  ...
}
```

`luaScriptAccessor`, `xxxScriptAccessor` can implement it.

Reply:
Sorry, I can understand clearly. We can only have one implementation `resourceCustomAccessor`, but it needs to implement all the interfaces.